2017/proposals/bansal-utkarsh.md

```diff
@@ -0,0 +1,628 @@
+
+    **Usage**
+
+    ```python
+    def f():
+      raise SystemExit(1)
+
+    def test_mytest():
+      with pytest.raises(SystemExit):
+        f()
+    ```
+
+5. `pytest.mark.xfail` to check test cases that are expected to fail. Usage already shown above.
+
+**Overall the project can be divided into three major parts which will cover issues [#884](https://github.com/MDAnalysis/mdanalysis/issues/884) Port to pytest and
+[#516](https://github.com/MDAnalysis/mdanalysis/issues/516) Update coordinate tests to use base test classes.**
+
+
+Part 1
+------
+This part will make the bare minimum changes needed to shift from nose to pytest and get TravisCI builds along with 
+code coverage running as usual. This is being done in a single PR to avoid any weird coexistence of two testing 
+libraries.
+
+* Modify the code to make it discoverable by pytest (take care of `__init__` and inheritance problems as explained above)
+* Port the 5 existing nose plugins to pytest. Pytest has awesome documentation on how to implement plugins which can be
+ found [here](http://doc.pytest.org/en/latest/writing_plugins.html) .
+    * Capture error (`capture_err.py`) This plugin captures stderr during test execution. If the test fails
+    or raises an error, the captured output is appended to the error or failure output. Pytest already has this
+feature inbuilt so this plugin can be dropped.
+
+        **Example**
+        ```python
+        def check_even(n, nn):
+            print(n)
+            assert n % 2 == 0 or nn % 2 == 0
+
+
+        def test_evens():
+            for i in range(0, 5):
+                yield check_even, i, i * 3
+        ```
+
+        **Output**
+        ```
+        ======================================= FAILURES =======================================
+        ____________________________________ test_evens[1] _____________________________________
+
+        n = 1, nn = 3
+
+            def check_even(n, nn):
+                print(n)
+        >       assert n % 2 == 0 or nn % 2 == 0
+        E       assert ((1 % 2) == 0 or (3 % 2) == 0)
+
+        experiments.py:37: AssertionError
+        --------------------------------- Captured stdout call ---------------------------------
+        1
+        ____________________________________ test_evens[3] _____________________________________
+
+        n = 3, nn = 9
+
+            def check_even(n, nn):
+                print(n)
+        >       assert n % 2 == 0 or nn % 2 == 0
+        E       assert ((3 % 2) == 0 or (9 % 2) == 0)
+
+        experiments.py:37: AssertionError
+        --------------------------------- Captured stdout call ---------------------------------
+        3
+        ```
+
+
+    * Cleanup (`cleanup.py`) This plugin deletes XDR offset files after all testing is done. This will have to
+    be re-written as a pytest plugin.
+    * Knownfailure (`knownfailure.py`) This is a decorator to check tests that are expected to fail. Pytest has
```

Comment:
I'll have a look at the plugin code once. But from how it is used in the tests, the function is the same.